content/open-source/wolfi/building-a-wolfi-package.md

```diff
@@ -0,0 +1,437 @@
+        mv "${{targets.destdir}}/$EXTENSIONS_DIR/${{range.key}}.so" \
+          "${{targets.subpkgdir}}/$EXTENSIONS_DIR/${{range.key}}.so"
+        prefix=
+        [ "${{range.key}}" != "opcache" ] || prefix="zend_"
+        echo "${prefix}extension=${{range.key}}.so" > $CONF_DIR/"${{range.key}}.ini"
+```
+
+And this will loop through all values of the `extensions` range and execute the described pipeline.
+
+#### The `update` Section
+This final section of the YAML file is only required when submitting the package to the Wolfi OS repository. The update section is used by Wolfi CI/CD systems to detect new package releases.
+
+Wolfi uses multiple tools and services to keep track of upstream releases, including the [Release Monitoring](https://release-monitoring.org/) service. For packages that are released via GitHub, tracking is made via the project org/name and a monitored tag.
+
+Here's an example of the update section of the PHP package, which uses the Release Monitoring service:
+
+```yaml
+update:
+  enabled: true
+  release-monitor:
+    identifier: 3627
+```
+The identifier can be obtained from the [release monitoring page](https://release-monitoring.org/) - search for the package and grab the ID that shows up at the URL.
+
+Here is another example, this time from a package that is released via GitHub:
+
+```yaml
+update:
+  enabled: true
+  github:
+    identifier: php-amqp/php-amqp
+    strip-prefix: v
+    tag-filter: v
+```
+Again, this section is only required when submitting the package to Wolfi.
+
+## Building the Packages
+When you feel your YAML is good to run, it's time to build the packages with melange. In this guide we'll use Docker to execute melange in a local environment. The procedure is explained in more detail in our [Getting Started with melange](https://edu.chainguard.dev/open-source/melange/tutorials/getting-started-with-melange/) tutorial.
+
+First, you'll need to generate a melange key. This is required to sign the apks, and the same key will need to be used when installing these packages in an image with apko.
+
+Make sure you have an updated version of the melange image:
+
+```shell
+docker pull cgr.dev/chainguard/melange
+```
+
+Then, run the following command to generate the key:
+
+```shell
+docker run --rm -v "${PWD}":/work cgr.dev/chainguard/melange keygen
+```
+
+Once you have the key, you can run the `build` command. A few explanations before you hit the keyboard:
+
+- **Building as root:** building packages require you to run melange as root, thus the `--privileged` parameter.
+- **Volumes:** here we set up **two** volumes: the current folder is shared with the /work location in the container, so that the packages generated will be available in your local system at the ./packages location. The second volume shares the `/tmp/melange_out` location in the container with your `/tmp` folder, so that you can have access to melange's working files even when the build fails. This helps a lot with debugging!
```

Comment:
```suggestion
- **Volumes:** here we set up **two** volumes: the current folder is shared with the /work location in the container, so that the packages generated will be available in your local system at the ./packages location. The second volume shares the `/tmp/melange_out` location in the container with your `/tmp` folder, so that you can have access to melange's working files even when the build fails. This helps a lot with debugging.
```